components/board/collector.go

```diff
@@ -26,78 +27,66 @@ func (m method) String() string {
 	return "Unknown"
 }
 
-// AnalogRecords a collection of AnalogRecord.
-type AnalogRecords struct {
-	Readings []AnalogRecord
-}
-
-// AnalogRecord a single analog reading.
-type AnalogRecord struct {
-	AnalogName  string
-	AnalogValue int
-}
-
-// GpioRecords a collection of GpioRecord.
-type GpioRecords struct {
-	Readings []GpioRecord
-}
-
-// GpioRecord a signle gpio reading.
-type GpioRecord struct {
-	GPIOName  string
-	GPIOValue bool
-}
-
-func newAnalogCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+// NewAnalogCollector returns a collector to register an analog reading method. If one is already registered
+// with the same MethodMetadata it will panic.
+func NewAnalogCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
```

Comment:
(fyi) I'll do this closer to the RDK release cut to coordinate with app, but we'll want to rename the above to the actual API methods: ReadAnalogReader and GetGPIO